spring-boot/proxy/src/test/java/eu/coatrack/proxy/security/local_api_key_manager_tests/ApiKeyFindingInLocalApiKeyListTest.java

```
@@ -0,0 +1,105 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import eu.coatrack.api.ApiKey;
+import eu.coatrack.proxy.security.exceptions.ApiKeyFetchingFailedException;
+import eu.coatrack.proxy.security.LocalApiKeyManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class ApiKeyFindingInLocalApiKeyListTest extends AbstractLocalApiKeyManagerSetup {
+
+    @BeforeEach
+    public void fillLocalApiKeyListWithListContainingValidApiKey() throws ApiKeyFetchingFailedException {
+        super.setupLocalApiKeyManagerAndApiKeyList();
+        when(apiKeyFetcherMock.requestLatestApiKeyListFromAdmin()).thenReturn(apiKeyList);
+        localApiKeyManager.updateLocalApiKeyList();
+    }
+
+    @Test
+    public void nullApiKeyValueShouldBeAnsweredWithNull() {
+        assertNull(localApiKeyManager.getApiKeyEntityByApiKeyValue(null));
+    }
+
+    @Test
+    public void apiKeyIsFoundInLocalApiKeyListAndThereforeReturned() {
+        assertSame(apiKey, localApiKeyManager.getApiKeyEntityByApiKeyValue(apiKey.getKeyValue()));
+    }
+
+    @Test
+    public void apiKeyIsNotFoundInLocalApiKeyListAndThereforeNullIsReturned() throws ApiKeyFetchingFailedException {
+        List<ApiKey> apiKeyListNotContainingTheIncomingApiKey = createApiKeyListNotContainingTheIncomingApiKey();
+
+        reset(apiKeyFetcherMock);
+        when(apiKeyFetcherMock.requestLatestApiKeyListFromAdmin()).thenReturn(apiKeyListNotContainingTheIncomingApiKey);
+        localApiKeyManager.updateLocalApiKeyList();
+
+        assertNull(localApiKeyManager.getApiKeyEntityByApiKeyValue(apiKey.getKeyValue()));
```

Should it maybe return an exception instead of null?

---

I think it is expected behavior to reject an incoming API key because of a missing right of access. Exceptions are therefore inappropriate.

---

I introduced an ApiKeyNotFoundInLocalApiKeyListException for this purpose.